pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -786,6 +788,11 @@ func (s *DiscoveryServer) pushEds(push *model.PushContext, con *XdsConnection, e
 			l = filteredCLA
 		}
 
+		// If location prioritised load balancing is enabled, prioritise endpoints.
+		if pilot.EnableLocalityLoadBalancing() {
+			loadbalancer.ApplyLocalityLBSetting(con.modelNode.Locality, l, s.Env.Mesh.LocalityLbSetting)
```

Comment:
@rshriram WDYT

Reply:
Wasn't this already enabled by default? I dont see a reason to require a flag to enable this commonly expected behavior.. As such, people perceive the lack of locality aware splaying as a bug. Also, this is not applicable to EDS as it stands today. So, I am more inclined to simply enable it by default. 

There are not many people enabling localities in service entries, except those using it in multicluster gateway scenarios.. like @aattuluri  and co. So this code will not get activated anyway (or shouldn't) until someone sets a locality.. Am I missing something here?

Reply:
The issue is the potential performance impact on Pilot. This is calculating locality relativity on the fly and not caching it and does this every request. This close to 1.1 we need stability so it's defaulting to off, but there are also a few big users that want to trial this functionality.

Reply:
@rshriram No, we only support strict-dns before this pr. Not do locality lb setting for eds endpoint

Reply:
I agree with the performance impact, but using flags in addition to config would make it difficult to discover/debug issues. Instead of the flag, can't we just move the check to see if `localityLBSettings` is configured to this level?

Reply:
`localityLBSettings` are specifically for overriding locality prioritised with either user-defined failover or weighting so can't be used in this case. 

This env var is just a temporary measure for 1.1 as is typical for experimental features with Pilot. Eventually, when we have had a couple of larger users test this and we have a better idea of the impact this causes the flag will go away and locality prioritised will be on by default.

Reply:
@rshriram localities are set in all Kube environments. This code uses localities retrieved in the kube platform code not the ones that can be sent as part of the xDS connection.